### PR TITLE
CRD changes

### PR DESCRIPTION
**Pull request by wanghaoran1988**
This change teh CRD by:
1. Adding UpgradeAt by allowing set when the upgrade should start
2. Adding Proceed by allowing a pre-configure upgrade can be canceled if
the upgrade haven't started.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ docker-build: build
 .PHONY: generate
 generate:
 	operator-sdk generate k8s
-	operator-sdk generate crds
+	operator-sdk generate crds --crd-version v1
```
Comment (dofinn) [inline Makefile:19]: v1 should be `v1alpha1` ?

File: deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml [hunk 2]
```diff
@@ -71,6 +71,11 @@ spec:
                 - force
                 - version
                 type: object
+              proceed:
+                default: true
+                description: Procceed whether to upgrade or not, this gives operator
+                  the ability to cancel the upgrade if upgrade hasn't started
```
Comment (mrbarge) [inline deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml:77]: Just very minor spelling/grammar fix
```suggestion
                description: Proceed whether to upgrade or not, this gives the operator
                  the ability to cancel the upgrade if the upgrade hasn't started
```

File: pkg/apis/upgrade/v1alpha1/upgradeconfig_types.go [hunk 3]
```diff
@@ -12,6 +12,13 @@ type UpgradeConfigSpec struct {
 	// Specify the desired OpenShift release
 	Desired Update `json:"desired"`
 
+	// Specify the upgrade start time
+	UpgradeAt string `json:"upgradeAt"`
+
+	// +kubebuilder:default:=true
+	// Procceed whether to upgrade or not, this gives operator the ability to cancel the upgrade if upgrade hasn't started
```
Comment (mrbarge) [inline pkg/apis/upgrade/v1alpha1/upgradeconfig_types.go:19]: Very minor spelling/grammar fix
```suggestion
	// Proceed whether to upgrade or not, this gives the operator the ability to cancel the upgrade if the upgrade hasn't started
```

File: deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml [hunk 4]
```diff
@@ -71,6 +71,11 @@ spec:
                 - force
                 - version
                 type: object
+              proceed:
+                default: true
+                description: Procceed whether to upgrade or not, this gives operator
```
Comment (dofinn) [inline deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml:76]: Can we add more reasoning to the description. Something like:
"Given all conditions have passed and cluster is ready to upgrade, proceed governs this decision to continue and commence the upgrade".

File: Makefile (hunk 1 above)
Comment (wanghaoran1988) [reply Makefile:19]: No, it should be v1, the first time I adding api are using this version, but I forget why.

File: deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml [hunk 5]
```diff
@@ -92,8 +97,13 @@ spec:
                   - namespace
                   type: object
                 type: array
+              upgradeAt:
+                description: Specify the upgrade start time
+                type: string
```
Comment (bmeng) [inline deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml:102]: should also define the format here?

Comment (wanghaoran1988) [reply deploy/crds/upgrade.managed.openshift.io_upgradeconfigs_crd.yaml:102]: it's a time stamp

**Pull request merged**